tests/sentry/backup/test_imports.py

```diff
@@ -529,11 +531,100 @@ def test_global_import_scoping(self):
 
         with tempfile.TemporaryDirectory() as tmp_dir:
             tmp_path = self.export_to_tmp_file_and_clear_database(tmp_dir)
-            with open(tmp_path) as tmp_file:
+            with open(tmp_path, "rb") as tmp_file:
                 import_in_global_scope(tmp_file, printer=NOOP_PRINTER)
                 self.verify_model_inclusion(ImportScope.Global)
 
 
+class DecryptionTests(ImportTestCase):
+    """
+    Ensures that decryption actually works. We only test one model for each scope, because it's
+    extremely unlikely that a failed decryption will leave only part of the data unmangled.
+    """
+
+    @staticmethod
+    def encrypt_json_fixture(tmp_dir) -> Tuple[Path, Path]:
+        good_file_path = get_fixture_path("backup", "fresh-install.json")
+        (priv_key_pem, pub_key_pem) = generate_rsa_key_pair()
+
+        tmp_priv_key_path = Path(tmp_dir).joinpath("key")
+        with open(tmp_priv_key_path, "wb") as f:
+            f.write(priv_key_pem)
+
+        tmp_pub_key_path = Path(tmp_dir).joinpath("key.pub")
+        with open(tmp_pub_key_path, "wb") as f:
+            f.write(pub_key_pem)
+
+        with open(good_file_path) as f:
+            data = json.load(f)
+
+        tmp_tarball_path = Path(tmp_dir).joinpath("input.tar")
+        with open(tmp_tarball_path, "wb") as i, open(tmp_pub_key_path, "rb") as p:
+            i.write(create_encrypted_export_tarball(data, p).getvalue())
```

Comment:
Is it worth it to have a testcase encrypting an export tarbell without this function `create_encrypted_export_tarbell`? Since technically changing the behavior in this function could lead to all these test cases failing.

Reply:
Good idea, I copied out the logic into this testing method.